### PR TITLE
i#1312 AVX-512 support: Provide ymm field in mcontext.

### PR DESCRIPTION
**Pull request by hgreving2304**
Provides a union with ymm field in addition to simd for backward compatibility. The name
was changed in f9c8364fe41752979788a35d.

Issue: #1312

### CODE REVIEW

File: core/lib/mcxtx.h [hunk 1]
```diff
@@ -256,5 +256,11 @@
      * bytes.
      */
 #    endif
-    dr_zmm_t simd[MCXT_NUM_SIMD_SLOTS];
+    union {
+        dr_zmm_t simd[MCXT_NUM_SIMD_SLOTS];
```
Comment (derekbruening) [inline core/lib/mcxtx.h]: These need their own doxygen comments as the existing one is now attached solely to the union.  You can see what the existing code above is doing: please follow that lead.  Feel free to update the existing comment lines 222-226 while doing that -- is that stale now?

File: core/lib/mcxtx.h [hunk 2]
```diff
@@ -256,5 +256,11 @@
...
+#    ifdef AVOID_API_EXPORT
+        /* The ymm field is provided for backward compat. */
```
Comment (derekbruening) [inline core/lib/mcxtx.h]: I would make this the exposed doxygen comment for this field and include "an alias for the `simd` field" or sthg so we don't have to duplicate the big comment on simd.

File: core/lib/mcxtx.h (hunk 1 above)
Comment (hgreving2304) [reply core/lib/mcxtx.h]: Done.
222-226 is not stale.

File: core/lib/mcxtx.h (hunk 2 above)
Comment (hgreving2304) [reply core/lib/mcxtx.h]: Done. I've also added a release note about the name change (# links do not apply to structure fields)

File: core/lib/mcxtx.h [hunk 3]
```diff
@@ -218,43 +218,51 @@
...
-     * that are not WOW64, or if the underlying processor does not support
-     * SSE.  Use dr_mcontext_xmm_fields_valid() to determine whether the
-     * fields are valid.
-     *
-     * When the fields are valid, on processors with AVX enabled (i.e.,
-     * proc_has_feature(FEATURE_AVX) returns true), these fields will
-     * contain the full ymm register values; otherwise, the top 128
-     * bits of each slot will be undefined.
-     */
+    union {
```
Comment (derekbruening) [inline core/lib/mcxtx.h]: My previous comment "You can see what the existing code above is doing: please follow that lead" implied also using a doxygen comment on the union like all the others are doing!

Comment (hgreving2304) [reply core/lib/mcxtx.h]: Done.

**Pull request merged**